target/linux/mediatek/filogic/base-files/etc/hotplug.d/ieee80211/11_fix_wifi_mac

```diff
@@ -16,6 +16,7 @@ case "$board" in
 		[ "$PHYNBR" = "1" ] && cat $key_path/6gMAC > /sys${DEVPATH}/macaddress
 		[ "$PHYNBR" = "2" ] && cat $key_path/5gMAC > /sys${DEVPATH}/macaddress
 		;;
+	asus,rt-ax59u|\
```

Comment:
I got wrong MACs with this code.
```
                :  Stock            : OpenWrt (this PR)
MAC address 2g	: 08:xx:xx:0D:xx:FE : 0A:xx:xx:0D:xx:FF
MAC address 5g	: 0A:xx:xx:1D:xx:FE : 08:xx:xx:0D:xx:FE
```

The working one for my device:
```
	asus,rt-ax59u)
		CI_UBIPART="UBI_DEV"
		addr=$(mtd_get_mac_binary_ubi "Factory" 0x4)
		[ "$PHYNBR" = "1" ] && \
			macaddr_setbit_la $(macaddr_setbit $addr 7) > /sys${DEVPATH}/macaddress
		;;
```

Reply:
I've tested another AX59U with original Asus firmwares and that is my MAC table:
```
                 :  Stock 3.0.0.4.288 31345 and 32702
MAC address 2g   : C8:7F:XX:XX:XX:BE
MAC address 5g   : CA:7F:XX:XX:XX:BE
MAC address LAN2 : C8:7F:XX:XX:XX:BE
MAC address WAN  : C8:7F:XX:XX:XX:BE
```

Maybe we should only use setbit_la in phy1

```
        asus,rt-ax59u)
                CI_UBIPART="UBI_DEV"
                addr=$(mtd_get_mac_binary_ubi "Factory" 0x4)
                [ "$PHYNBR" = "1" ] && \
                        macaddr_setbit_la $addr > /sys${DEVPATH}/macaddress
                ;;

```

Reply:
> Maybe we should only use `setbit_la` in phy1

It's possible. The flaw is that the 2nd(3rd, ...) 2.4 SSID will be equal to the 1st(2nd, ...) 5 GHz SSID

> : Stock 3.0.0.4.288 31345 and 32702
> MAC address 2g   : C8:7F:XX:XX:XX:BE
> MAC address 5g   : CA:7F:XX:XX:XX:BE
> MAC address LAN2 : C8:7F:XX:XX:XX:BE
> MAC address WAN  : C8:7F:XX:XX:XX:BE

Could you show the 4th byte of the MAC, please? I also checked both 3.0.0.4.388_31345 and 3.0.0.4.388_32702.

```
                                :  Stock
MAC address 2g	: 08:xx:xx:0D:xx:FE
MAC address 5g	: 0A:xx:xx:1D:xx:FE
LAN             : 08:xx:xx:0D:xx:FE
WAN             : 08:xx:xx:0D:xx:FE
```

We need more statistics. @tkso1997 can you share your MACs, please?

Reply:
```
                 : Stock 3.0.0.4.288 31345 and 32702
MAC address 2g   : C8:7F:XX:76:XX:BE
MAC address 5g   : CA:7F:XX:76:XX:BE
MAC address LAN2 : C8:7F:XX:76:XX:BE
MAC address WAN  : C8:7F:XX:76:XX:BE
```

I have more routers here, I'm going to check some more routers and will post results here.

Reply:
@xavifr, thanks. Mow everything is clear. C8:7F:XX:76:XX:BE already has 1 in 28th bit. As a result the 4th bytes are identical.

My proposal is:
1. Keep original ASUS algorithm (and set 28th bit);
2. Increase by one the 5g MAC to avoid 2g and 5g MAC addresses matching (this should be documented in the commit message). You 5g MAC will be `C8:7F:XX:76:XX:BF`.
```
	asus,rt-ax59u)
		CI_UBIPART="UBI_DEV"
		addr=$(mtd_get_mac_binary_ubi "Factory" 0x4)
		[ "$PHYNBR" = "1" ] && \
			macaddr_setbit_la $(macaddr_setbit $(macaddr_add $addr 1) 28) > \
			/sys${DEVPATH}/macaddress
		;;

```

Reply:
I can confirm your thoughts, 

```
                 : Stock 3.0.0.4.288 31345
MAC address 2g   : 08:BF:XX:0D:XX:56
MAC address 5g   : 0A:BF:XX:1D:XX:56
MAC address LAN2 : 08:BF:XX:0D:XX:56
MAC address WAN  : 08:BF:XX:0D:XX:56
```